src/foam/dao/DAOProperty.js

```diff
@@ -42,10 +42,16 @@ foam.CLASS({
     },
     {
       name: 'adapt',
-      value: function(_, o) {
+      value: function(_, o, prop) {
         if ( ! o ) return;
         if ( foam.String.isInstance(o) && this.__subContext__ ) return this.__subContext__[o];
-        return o;
+
+        var type = foam.lookup(prop.type);
```

Comment:
Isn't this the same as FObjectProperty's adapt()? Couldn't we just call that rather than duplicating the code?